syntax  = "proto3";
package simpledb.proto;

message DictArg {
    string key = 1;
    bytes val = 2;
    bool is_file = 3;
};

message ListArg {
    bytes val = 1;
    bool is_file = 2;
};

message CPPExecArgs {
    repeated ListArg args = 1;
    repeated DictArg kwargs = 2;
};

message CPPExecResponse {
    int32 return_code = 1;
    bytes output = 2;
};